Add tooltip to GameTableController

Katharsas opened the pull request:
Same tooltip controller as used in GameTileController.
Fixes #1338
Fixes #804 

May need some help with tests if those are wanted and if its even possible to test thist.

Katharsas committed:
Add tooltip to GameTabelController

Same tooltip controller as used in GameTileController.
Fixes #1338

1-alex98 commented:
Can u post a screen shot so I can imagine what u are doing here?

Katharsas commented:
![coop_tooltip](https://user-images.githubusercontent.com/1922573/60610424-1f8d1000-9dc4-11e9-8f84-8198853465da.JPG)
(Same for table view of custom games. I can restrict it to only show hen mouse hovers over selected row, but right now it shows tooltips for any row hen hovered over).

1-alex98 commented:
This is now also the case in the table of normal games right....

1-alex98 commented:
I am on discord and we can try to figure out how to test it...

Katharsas commented:
yeah thats what i meant.

Brutus5000 reviewed (commented):
I'm not convinced by this change yet. Why do we need to reload the tooltip every time from the fxml? I guess this is pretty expensive. Why not store one loaded controller and just update the displayed game?

Katharsas committed:
Fixing various things

- Potentially improved performance by only having one Tooltip and GameTooltipController per game list / game tile list
- Added space between teams inside tooltip (wrapping vbox in vbox was necessary for this to work)
- Made teams inside tiles have same size always to prevent weird layout
- Fixed Tooltip spacing problem (new helper method for adding content to a ToolTip inside JavaFxUtil)
- Fixed Tooltip spacing when only one team exists
- Fixed exceptions being thrown by GamesTableControllerTest

1-alex98 commented:
We fixed this along with many other things concerning the tool tip a second ago

Katharsas committed:
Fixed tests

Katharsas committed:
remove unused imports

1-alex98 commented on src/main/java/com/faforever/client/game/GamesTableController.java:
```diff
@@ -103,7 +110,7 @@ public void initializeGameTable(ObservableList<Game> games, Function<String, Str
     applyLastSorting(gamesTable);
     gamesTable.setOnSort(this::onColumnSorted);
-
+    
```
?

1-alex98 commented on src/test/java/com/faforever/client/game/GamesTableControllerTest.java:
```diff
@@ -50,20 +65,26 @@ public void setUp() throws Exception {
-        GameBuilder.create().defaultValues().get(),
-        GameBuilder.create().defaultValues().state(GameStatus.CLOSED).get()
-    ));
+    Platform.runLater(() -> {
```
Use WaitForAsyncUtils.executeAsyncFX something

1-alex98 replied:
and then get()

Katharsas committed:
Remove spaces

Brutus5000 commented:
Looks better to me now, but I'm not deep into the JavaFX stuff to actually approve this. It's up to @axel1200 now.
